app/controllers/saml_idp_controller.rb

```diff
@@ -7,14 +7,16 @@ class SamlIdpController < ApplicationController
   include SamlIdpAuthConcern
   include SamlIdpLogoutConcern
   include FullyAuthenticatable
+  include VerifyProfileConcern
 
   skip_before_action :verify_authenticity_token
   skip_before_action :handle_two_factor_authentication, only: :logout
 
   def auth
     return confirm_two_factor_authenticated(@request_id) unless user_fully_authenticated?
-    process_fully_authenticated_user do |needs_idv|
-      return store_location_and_redirect_to_verify_url if needs_idv
+    process_fully_authenticated_user do |needs_idv, needs_profile_finish|
+      return store_location_and_redirect_to(verify_url) if needs_idv
+      return store_location_and_redirect_to(profile_or_verify_profile_url) if needs_profile_finish
```

Comment:
Yep, and that logic comes from `verify_profile_route`, right? I'm saying call the `verify_profile_route` method directly instead of going through the logic in `profile_or_verify_profile_url`, which will always return `verify_profile_route` in this scenario.